marlowe-playground-client/src/SimulationPage/Types.purs

```diff
@@ -132,6 +135,7 @@ type ExecutionStateRecord
     , state :: S.State
     , slot :: Slot
     , moneyInContract :: Assets
+    , contract :: S.Contract
```

Comment:
This gets stored for each step of the simulation?

Reply:
I cannot think of an alternative if we want to be able to undo. Unless we start passing a Zipper to the semantics or something like that. In Haskell at least, I would expect that it doesn't require extra memory, because values are immutable, so they should not be copies, just pointers to subtrees of the original Contract. Not sure if that applies to Purescript